discovery.go

```diff
@@ -5,99 +5,114 @@ import (
 	"sync"
 	"time"
 
+	"github.com/Sirupsen/logrus"
 	consulapi "github.com/hashicorp/consul/api"
 )
 
-// Short description
-type Address struct {
-	Ip   string
-	Port int
+type NameProvider interface {
+	// Returns a name for the service or an empty string,
+	// if no name could be determined.
+	GetName(host string, port uint16) string
 }
 
-// Short Description
-var consulServices map[Address]string = make(map[Address]string)
-var mutex sync.Mutex
-var firstUpdate sync.Once
+type NoopNameProvider struct{}
 
-// Short description
-func getServices(consul *consulapi.Client) map[string][]string {
-	catalog := consul.Catalog()
+func (*NoopNameProvider) GetName(host string, port uint16) string {
+	return ""
+}
 
-	results, _, err := catalog.Services(nil)
-	if err != nil {
-		return nil
-	}
+// Short description
+type consulAddressKey struct {
```

Comment:
do we need the node address to to make it unique? we need to avoid having agents reporting to the same kafka running on hosts with docker daemons that might create the same ip addresses. remember the loadbalancer issue in live that we had with the endless loop?